File: src/types/strategy.ts
import type { SignalTagType } from './stock';

/** 策略类型 */
export type StrategyType = '低吸' | '趋势' | '突破' | '均值回归';

/** 会员等级 */
export type PlanLevel = 'free' | 'pro' | 'flagship';

/** 风格预设 */
export type StylePreset = 'conservative' | 'balanced' | 'aggressive';

/** 策略 */
export interface Strategy {
  id: string;
  name: string;
  type: StrategyType;
  description: string;
  total_return: number;
  win_rate: number;
  max_drawdown: number;
  sharpe_ratio: number;
  sortino_ratio: number;
  calmar_ratio: number;
  running_days: number;
  signal_count: number;
  required_plan: PlanLevel;
  is_subscribed: boolean;
}

/** 预警设置 */
export interface AlertSetting {
  stop_loss_pct: number;
  take_profit_pct: number;
  alert_method: 'notification' | 'sms' | 'wechat';
  alert_frequency: 'once' | 'daily' | 'every';
}

/** 因子标签 */
export type FactorLabel =
  | '估值优势'
  | '涨势动力'
  | '资金热度'
  | '反弹潜力'
  | '市场温度'
  | '震荡程度'
  | '突破强度';

/** 因子状态 */
export interface FactorState {
  label: FactorLabel;
  value: number;
  change: number;
  status: 'normal' | 'warning' | 'danger';
}

File: backend/schemas/strategy.py
"""策略相关 Pydantic Schema

包含：StrategyOut, FactorStateOut, PerformanceOut
"""

from pydantic import BaseModel, Field


class StrategyOut(BaseModel):
    """策略详情输出"""

    id: str
    name: str
    type: str
    description: str | None = None
    total_return: float = 0.0
    win_rate: float = 0.0
    max_drawdown: float = 0.0
    sharpe_ratio: float = 0.0
    sortino_ratio: float = 0.0
    calmar_ratio: float = 0.0
    running_days: int = 0
    signal_count: int = 0
    required_plan: str = "free"
    is_active: bool = True
    is_subscribed: bool = False

    model_config = {"from_attributes": True}


class FactorStateOut(BaseModel):
    """因子状态输出"""

    label: str
    value: float = 0.0
    change: float = 0.0
    status: str = "neutral"

    model_config = {"from_attributes": True}


class PerformanceOut(BaseModel):
    """策略绩效曲线输出"""

    dates: list[str] = Field(default_factory=list, description="日期序列")
    strategy_values: list[float] = Field(default_factory=list, description="策略净值序列")
    benchmark_values: list[float] = Field(default_factory=list, description="基准净值序列")
    total_return: float = 0.0
    max_drawdown: float = 0.0
    sharpe_ratio: float = 0.0
    trades: list[dict] = Field(default_factory=list, description="交易记录列表")
